src/SonarScanner.MSBuild.PreProcessor/FilePermissionsWrapper.cs

```diff
@@ -33,35 +33,25 @@ public void Set(string destinationPath, int mode)
     {
         if (operatingSystemProvider.IsUnix())
         {
-            if (operatingSystemProvider.OperatingSystem() is PlatformOS.Alpine)
+            // https://github.com/Jackett/Jackett/blob/master/src/Jackett.Server/Services/FilePermissionService.cs#L27
+            var process = new Process
             {
-                // https://github.com/Jackett/Jackett/blob/master/src/Jackett.Server/Services/FilePermissionService.cs#L27
-                var process = new Process
+                StartInfo = new ProcessStartInfo
                 {
-                    StartInfo = new ProcessStartInfo
-                    {
-                        RedirectStandardError = true,
-                        UseShellExecute = false,
-                        CreateNoWindow = true,
-                        WindowStyle = ProcessWindowStyle.Hidden,
-                        FileName = "chmod",
-                        Arguments = $"""{Convert.ToString(mode, 8)} "{destinationPath}" """,
-                    }
-                };
-                process.Start();
-                var stdError = process.StandardError.ReadToEnd();
-                process.WaitForExit();
-                if (process.ExitCode != 0)
-                {
-                    throw new InvalidOperationException(stdError);
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    WindowStyle = ProcessWindowStyle.Hidden,
+                    FileName = "chmod",
+                    Arguments = $"""{Convert.ToString(mode, 8)} "{destinationPath}" """,
                 }
-            }
-            else
+            };
+            process.Start();
+            var stdError = process.StandardError.ReadToEnd();
+            process.WaitForExit();
```

Comment:
To speed up things, we could:
* Make `Set` async
* Use `WaitForExitAsync` here.
* Call `SetAsync` in the unpacked and store the Task in a list.
* `await` all the permission setting tasks with `await Task.WhenAll(list)` after all files are extracted.

To enable this, the `IUnpacker.Unpack` needs to be made async. I don't think we gain much, and we may have a problem when we spawn too many processes in parallel. I don't think the gain is worth the risk here. What do you think?

Reply:
I think as well that there is not much gain, as the bottleneck is the JRE download. 
Take a look at the timestamps on my ubuntu test runs:

```md
10:25:29.424  Downloading Java JRE from https://scanner.sonarcloud.io/jres/OpenJDK17U-jre_x64_linux_hotspot_17.0.11_9.tar.gz.
##### 10 seconds for the download ^

10:25:38.57  The checksum of the downloaded file is 'bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707' and the expected checksum is 'bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707'.
10:25:38.571  Starting extracting the Java runtime environment from archive '/home/greg/.sonar/cache/bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707/OpenJDK17U-jre_x64_linux_hotspot_17.0.11_9.tar.gz' to folder '/home/greg/.sonar/cache/bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707/dtg4eiml.ofr'.
##### Almost instant for the extraction+permissions setting. %

10:25:39.665  Moving extracted Java runtime environment from '/home/greg/.sonar/cache/bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707/dtg4eiml.ofr' to '/home/greg/.sonar/cache/bcb1b7b8ad68c93093f09b591b7cb17161d39891f7d29d33a586f5a328603707/OpenJDK17U-jre_x64_linux_hotspot_17.0.11_9.tar.gz_extracted'.
````